core/src/main/scala/org/apache/spark/sql/crossdata/catalog/CatalogChain.scala

```diff
@@ -72,6 +115,7 @@ private[crossdata] class CatalogChain private(val temporaryCatalogs: Seq[XDTempo
     temporaryCatalogs.foreach(_.saveView(viewIdentifier, logicalPlan, sql))
 
   // TODO throw an exception if there is no temp catalogs! Review CatalogChain
```

Comment:
You could use `require` at `CatalogChain` constructor, or even here, to enforce it.